bitcoin/src/blockdata/script/mod.rs

```diff
@@ -206,11 +193,46 @@ pub fn read_scriptint(v: &[u8]) -> Result<i64, Error> {
 /// See [`read_scriptint`] for a description of some subtleties of
 /// this function.
 pub fn read_scriptint_non_minimal(v: &[u8]) -> Result<i64, Error> {
+    read_scriptint_size(v, DEFAULT_MAX_SCRIPTINT_SIZE, false)
+}
+
+/// Decodes an interger in script format with flexible size limit.
+///
+/// Note that in the majority of cases, you will want to use either
+/// [read_scriptint] or [read_scriptint_non_minimal] instead.
+///
+/// Panics if max_size exceeds 8.
+pub fn read_scriptint_size(v: &[u8], max_size: usize, minimal: bool) -> Result<i64, Error> {
```

Comment:
Should be a method on some type. And does it really have to be public?

Reply:
@Kixunil I'm not sure if this PR is being maintained, but to understand how it's used you'll want to poke into the bitvm repos. In particular https://github.com/BitVM/rust-bitcoin-script

It's been a while since I looked at this but I remember being pretty surprised at how rich a script API they needed.

Reply:
It looks like they now point to the `bitvm` branch of this repo, which I think @tcharding made and points to the tip  of #2613.